src/schema/me/conversation/message.js

```diff
@@ -51,11 +51,17 @@ export const MessageType = new GraphQLObjectType({
       description: "True if message is from the user to the partner.",
       type: GraphQLBoolean,
       resolve: (
-        { from_id, from_email_address, conversation_from_address },
+        {
+          from_id,
+          from_email_address,
+          conversation_from_address,
+          from_principal,
+        },
         options,
         req,
         { rootValue: { userID } }
       ) =>
+        from_principal ||
```

Comment:
This will end up as `false` if `from_principal` is `null`, but it should probably return `null`.

Reply:
but since its an `or` we end up testing other scenarios right?

Reply:
It could still end up being wrong though, if it is from the collector side but not from the original sender. Do we really wanna try to guess here or just return nil for legacy?

Reply:
the type is `GraphQLBoolean` i _think_ we need to return a boolean, basically for old style conversations we do what we used to do which is email matching but as you mentioned there are cases that it can be wrong but i _think_ its ok to leave it as it is since we are not improving much on _classic conversations_ :)

Reply:
If `from_principal` is `null`, what does that mean exactly (on the Impulse side)?

Reply:
it means we don't know their side since this conversation was created before our changes where we added `principal`.

Reply:
Does that even get returned in the Impulse presenter though? I recall something where (either Radiation or Impulse) just didn't include a value that was `nil` (so it was missing from the response altogether).

Otherwise this makes sense to fallback to a the current 'guess' I think 👍

Reply:
What @saolsen said makes sense too. I guess in that case it's more of a chore of still supporting this for older messages, in that case might it make sense to do a migration on Impulse (and set it for messages where the `from` match the conversation originator)?

Reply:
@saolsen im not sure what is the impact on Eigen if we start sending `null` back and if we need to change `eigen` to handle `null` but even in the case we return `null` Eigen needs to show something for sender and most likely `Partner` name is better choice in case we don't know which side this message belongs to, it covers more cases.

@mzikherman i think presenter will exclude that field if its `nil` is that an issue? do we need to handle that case differently?

Reply:
Aight, we def have to do some make believe pretend games somewhere, I guess this is a good a place as any. We should really try to clean this all up in a couple months or so when all conversations are of the new style.

Reply:
totally agree, I don't think we can do a proper migration for existing conversations and messages but it will be less of an issue in few months 👍

Reply:
Sounds like the `null` case needs a spec to clarify this.

Reply:
@ashkan18 I just meant that if it's missing from the presenter its just slightly different than being present (and potentially `null`), but we got rid of all those `isTypeOf` checks a while back (where it was expecting certain fields to be present in the schema), so I think this is fine 👍